scripts/uploader.rb

```diff
@@ -1,46 +1,102 @@
-require 'fog'
+require 'active_support/all'
+require 'json'
+require 'rest-client'
+require 'yaml'
 
 module Build
   class Uploader
-    attr_reader :directory, :type
+    attr_reader :container, :directory, :type
 
     def self.upload(directory, type)
       new(directory, type).run
     end
 
     def initialize(directory, type)
+      @container = "release"
       @directory = directory
       @type      = type
     end
 
-    def service
-      # Example config/upload.yml
-      # ---
-      # :provider: XYZ
-      # :rackspace_username: MyUser
-      # :rackspace_api_key: MyKey
-      # :rackspace_region: :my_region
-      @service ||= Fog::Storage.new(YAML.load_file("#{__dir__}/../config/upload.yml"))
-    end
-
-    def container
-      @container ||= service.directories.get("release")
-    end
-
     def run
       appliances = Dir.glob("#{directory}/manageiq*")
       appliances.each do |appliance|
         # Skip badly named appliances, missing git sha: manageiq-openstack-master-201407142000-.qc2
-        next unless appliance.match(/.+-[0-9]{12}-[0-9a-fA-F]+/)
-        key = uploaded_filename(appliance)
-        puts "Uploading #{appliance} as #{key}..."
-        upload_file = File.expand_path(appliance)
-        file = container.files.create :key => key, :body => File.open(upload_file, "r")
-        url = file.public_url if file
-        puts "Uploading #{appliance} as #{key}...complete: #{url}"
+        unless appliance.match(/.+-[0-9]{12}-[0-9a-fA-F]+/)
+          puts "Skipping #{appliance}"
+          next
+        end
+
+        destination_name = uploaded_filename(appliance)
+        source_name = File.expand_path(appliance)
+
+        puts "Uploading #{appliance} as #{destination_name}..."
+
+        destination_url = url(destination_name)
+        source_hash = Digest::MD5.file(source_name).hexdigest
+
+        upload_headers = headers.merge("ETag" => source_hash)
+        unless release?
+          image_date = destination_name.split("-")[-2]
+          delete_at  = (DateTime.parse(image_date) + 8.weeks)
```

Comment:
Minor but can we make the `8.weeks` a constant?  That way if we want to change it, it's clearer where to find it.